### PR TITLE
nixos/stage1: run fsck on battery too

### PR DESCRIPTION
**Pull request by Artturin**
We are in 2021 almost 2022 not in 2004 when this may have been an issue!

https://alioth-lists.debian.net/pipermail/pkg-sysvinit-devel/2009-May/003196.html
https://www.nico.schottelius.org/blog/debian-ubuntu-fsck-skip-on-battery-bug/
https://github.com/OpenRC/openrc/commit/d29daf395299fc97b8e13676bc282800a8bddae8
https://bugs.launchpad.net/ubuntu/+source/laptop-mode/+bug/11194

###### Motivation for this change
The code was added in 2008! 477ea7015c8e50dc07fcb20af67c22152cf94470

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Lassulus — APPROVED** [review]

**Comment by samueldr** [comment]
In an ideal world, we'd probably instead look for a "good enough" charge level and ask with a timeout if the user wants to continue with fsck anyway. Let's say under 30% of power.

But I don't want to move the goalposts on that. Don't take this as a rejection, but a way forward if other maintainers don't like the idea of removing this.

**Comment by pennae** [comment]
:+1: for removing the restriction.

has ext4 moved away from periodic checks? if so there's *really* no reason to skip fsck ever.

**Comment by Artturin** [comment]
> has ext4 moved away from periodic checks? if so there's _really_ no reason to skip fsck ever.

i think so, at least on my raspberry pi's ext4 partition has `Maximum mount count` as `-1` which means its disabled
`sudo tune2fs -l  /dev/sda2 | grep Maximum`


> -c max-mount-counts
>        Adjust  the  number  of mounts after which the file system will be checked by
>        e2fsck(8).  If max-mount-counts is the string "random", tune2fs  will  use  a
>        random  value  between 20 and 40.  If max-mount-counts is 0 or -1, the number
>        of times the file system is mounted will be disregarded by e2fsck(8) and  the
>        kernel. 
>        Staggering  the  mount-counts at which file systems are forcibly checked will
>        avoid all file systems being checked at one time when  using  journaled  file
>        systems.
>        **Mount-count-dependent  checking is disabled by default to avoid unanticipated
>        long reboots while e2fsck does its work.**  If you  are  concerned  about  file
>        system  corruptions  caused  by potential hardware problems of kernel bugs, a
>        better solution than mount-count-dependent checking is to use the  e2scrub(8)
>        program.   This  does  require placing the file system on an LVM volume, how‐
>        ever.

**Comment by pennae** [comment]
> weren't the long fsck's mainly a ext2 or ext3 problem instead of ext4?

think so. didn't intend to reference fsck speed either, just that skipping fsck on boot is a thing that would've mostly benefited extN (and others that do periodic fscks when nothing looks wrong). if they don't do it anymore we needn't keep that optimization around either.

**Comment by mohe2015** [comment]
> In an ideal world, we'd probably instead look for a "good enough" charge level and ask with a timeout if the user wants to continue with fsck anyway. Let's say under 30% of power.
> 
> But I don't want to move the goalposts on that. Don't take this as a rejection, but a way forward if other maintainers don't like the idea of removing this.

I think one of the referenced issues just said that if you are not at home (so on battery) you don't want to wait for the fsck if you e.g. want to show somebody something so I think charge level wasn't even an argument

**Comment by samueldr** [comment]
> I think one of the referenced issues just said that if you are not at home (so on battery) you don't want to wait for the fsck if you e.g. want to show somebody something so I think charge level wasn't even an argument

Yeah, I didn't make this clear enough: *if the feature is still desired* by some maintainers, and they block this change. I personally agree.

**Pull request merged**
